go/vt/vtgate/planbuilder/operators/join.go

```diff
@@ -89,7 +74,13 @@ func (j *Join) Compact(ctx *plancontext.PlanningContext) (Operator, *ApplyResult
 
 func createStraightJoin(ctx *plancontext.PlanningContext, join *sqlparser.JoinTableExpr, lhs, rhs Operator) Operator {
 	// for inner joins we can treat the predicates as filters on top of the join
-	joinOp := &Join{LHS: lhs, RHS: rhs, JoinType: join.Join}
+	joinOp := &Join{
+		binaryOperator: binaryOperator{
+			LHS: lhs,
+			RHS: rhs,
+		},
+		JoinType: join.Join,
+	}
```

Comment:
We have new utils function like `newDistinct`, `newLockAndComment` etc, I am wondering if we can extend that to more operators, like `Join`, `HashJoin` etc. This would make the code more consistent.

Reply:
Join has constructor methods `createJoin`, `createLeftOuterJoin` and `createStraightJoin`. HashJoin has `NewHashJoin`